backend/src/test/java/com/google/sps/EmailNotifierTest.java

```diff
@@ -0,0 +1,177 @@
+package com.google.sps;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.model.Message;
+import com.google.common.truth.Truth;
+import com.google.sps.notifs.EmailNotifier;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.Properties;
+import javax.mail.Address;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import org.apache.commons.codec.binary.Base64;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+
+@RunWith(JUnit4.class)
+public class EmailNotifierTest {
+  private EmailNotifier emailNotifier;
+  @Mock private Gmail gmail;
+  @Mock private Gmail.Users users;
+  @Mock private Gmail.Users.Messages messages;
+  @Mock private Gmail.Users.Messages.Send send;
+  private Message testMessage;
+
+  @Before
+  public void setUp() throws MessagingException, IOException {
+    gmail = mock(Gmail.class);
+    // Mock to return list of users
+    users = mock(Gmail.Users.class);
+    // Mock to return list of messages.
+    messages = mock(Gmail.Users.Messages.class);
+    send = mock(Gmail.Users.Messages.Send.class);
+    MimeMessage mimemessage = createTestMessage();
+    testMessage = createMessageFromMime(mimemessage);
+    // Test Instance of email notifier class.
+    emailNotifier = new EmailNotifier("John", "jdoe@gmail.com ", gmail);
+    // Emulating real behavior due to when method, will return list of users.
+    when(gmail.users()).thenReturn(users);
+    // Emulating real behaviour due to when method, will return list of messages
+    when(users.messages()).thenReturn(messages);
+    //    Emulating sending messages, when sent will return send mock.
+    when(messages.send(any(), any())).thenReturn(send);
+    //    When send mock is executed will return test message.
+    when(send.execute()).thenReturn(testMessage);
```

Comment:
Since the tests now use the message constructed in EmailNotifier, can this line be removed?

Reply:
Deleted line 59-60